src/libjasper/jpc/jpc_t1cod.c

```diff
@@ -419,7 +421,9 @@ int jpc_getscctxno(int f)
 		} else {
 			n = 1;
 		}
-	} else if (hc == 1) {
+	} else {
+		assert(hc == 1);
```

Comment:
I recommend adding some error reporting in case of release configurations

Reply:
No, this is an impossible case. There's no point in catching such things in release builds - or else you'd have millions and billions of impossible cases everywhere else to check.
This is not about input validation!
This is about adding two boolean values and subtracting one. The logically possible output range is `-1..1`, and at this point, `-1` has been converted to `1` already.

Reply:
But wait... maybe I've made a mistake here, because thinking about it again, is `2` possible? Thanks for the comment, it may help me find a bug in my patch :-)

Reply:
I just read the code again - this just subtracts one bool from another. So the possible range is really `-1..1`.
The code is formatted very badly and expressed in a very obscure way, making it very hard to understand what is really being done.
But I'm confident that my change is correct. But it's always good to have somebody else double-check it....